src/main/scala/com/campudus/tableaux/database/domain/column.scala

```diff
@@ -64,105 +103,108 @@ sealed trait ColumnType[+A] extends DomainObject {
   def checkValidValue[B](value: B): Option[String] = if (value == null) None else kind.checkValidValue(value)
 }
 
-sealed trait ValueColumn[+A] extends ColumnType[A]
-
-/*
- * Single-language column types
- */
-sealed trait SimpleValueColumn[+A] extends ValueColumn[A]
-
-case class TextColumn(override val kind: TableauxDbType)(override val table: Table, override val id: ColumnId, override val name: String, override val ordering: Ordering, override val identifier: Boolean, override val displayInfos: Seq[DisplayInfo], override val countryCodes: Option[Seq[String]]) extends SimpleValueColumn[String]
-
-case class NumberColumn(table: Table, id: ColumnId, name: String, ordering: Ordering, override val identifier: Boolean, override val displayInfos: Seq[DisplayInfo], override val countryCodes: Option[Seq[String]]) extends SimpleValueColumn[Number] {
-  override val kind = NumericType
+/**
+  * Helper for pattern matching
+  */
+object MultiLanguageColumn {
+  def unapply(columnType: ColumnType[_]): Option[SimpleValueColumn[_]] = {
+    (columnType, columnType.languageType) match {
+      case (simpleValueColumn: SimpleValueColumn[_], _: MultiLanguage | _: MultiCountry) => Some(simpleValueColumn)
+      case _ => None
+    }
+  }
 }
 
-case class CurrencyColumn(override val table: Table, override val id: ColumnId, override val name: String, override val ordering: Ordering, override val identifier: Boolean, override val displayInfos: Seq[DisplayInfo], override val countryCodes: Option[Seq[String]]) extends SimpleValueColumn[Number] {
-  override val kind = CurrencyType
-}
+object SimpleValueColumn {
+  def apply(kind: TableauxDbType, languageType: LanguageType, columnInformation: ColumnInformation): SimpleValueColumn[_] = {
+    val applyFn: (LanguageType) => (ColumnInformation) => SimpleValueColumn[_] = kind match {
```

Comment:
And now I realized that the two types are different (TextTYPE -> TextCOLUMN.apply). Sorry, ignore this here ;)